src/Neo/SmartContract/Native/LedgerContract.cs

```diff
@@ -53,13 +54,29 @@ internal override ContractTask OnPersist(ApplicationEngine engine)
                 {
                     var conflictRecord = engine.Snapshot.GetAndChange(CreateStorageKey(Prefix_Transaction).Add(attr.Hash),
                         () => new StorageItem(new TransactionState { ConflictingSigners = Array.Empty<UInt160>() })).GetInteroperable<TransactionState>();
-                    conflictRecord.ConflictingSigners = conflictRecord.ConflictingSigners.Concat(conflictingSigners).Distinct().ToArray();
+                    conflictRecord.ConflictingSigners = conflictRecord.ConflictingSigners.Concat(conflictingSigners)
+                        .Distinct().OrderBy(s => s).Take(MaxAllowedConflictingSigners).ToArray();
```

Comment:
consider the case here:

<https://github.com/neo-project/neo/issues/2907#issuecomment-1722596148>